src/NUnitFramework/framework/Internal/Reflect.cs

```diff
@@ -113,11 +142,42 @@ internal static IEnumerable<ConstructorInfo> GetConstructors(Type type, Type?[]
         /// </summary>
         internal static bool ParametersMatch(this ParameterInfo[] pinfos, Type?[] ptypes)
         {
-            if (pinfos.Length != ptypes.Length)
-                return false;
+            bool hasParamsArgument = pinfos.Length > 0 && pinfos[pinfos.Length -1].HasAttribute<ParamArrayAttribute>(false);
+
+            if (hasParamsArgument)
+            {
+                if (ptypes.Length < pinfos.Length -1)
+                    return false;
+            }
+            else
+            {
+                if (pinfos.Length != ptypes.Length)
+                    return false;
+            }
 
             for (int i = 0; i < pinfos.Length; i++)
             {
+                if (hasParamsArgument && i == pinfos.Length - 1)
+                {
+                    var elementType = pinfos[i].ParameterType.GetElementType();
+                    if (elementType is not null)
+                    {
+                        bool allParamArgsMatched = true;
+                        for (int j = i; j < ptypes.Length; j++)
+                        {
+                            if (!ptypes[j].CanImplicitlyConvertTo(elementType))
+                            {
+                                allParamArgsMatched = false;
+                                break;
```

Comment:
This can be simplified by simply calling `return false` here. Then you don't need the `allParamArgsMatched` variable at all.
Keep the comment below in mind so you also deal with the case where the last parameter is an array instead of a set of values.

Reply:
Unfortunately I don't think this ends up being easier due to the problems with switching the order

Reply:
@Shiney I looked at it a bit more and you are correct.